afs-core/src/main/java/com/powsybl/afs/TaskMonitor.java

```diff
@@ -7,17 +7,29 @@
 package com.powsybl.afs;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 import java.util.Objects;
 import java.util.UUID;
+import java.util.concurrent.Future;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public interface TaskMonitor extends AutoCloseable {
 
+    class NotACancellableTaskMonitor extends Exception {
```

Comment:
Exception vs RuntimeException vs PowsyblException

Reply:
I don't want it  to be a RuntimeException so the user is aware that the taskmonitor may not be able to set the future to the task (in case of a remote task monitor) and explicitly catch the execption